amethyst_core/src/transform/components/transform.rs

```diff
@@ -585,6 +585,15 @@ impl Transform {
 
         res
     }
+
+    /// This is only exposed for debug builds, with `debug_assertions` set.
+    /// This function allows for test cases of copying the local matrix to the global matrix.
+    /// Useful for tests or other debug type access.
+    #[cfg(debug_assertions)]
+    #[inline]
+    pub fn copy_local_to_global(&mut self) {
```

Comment:
Is this meant to be part of the public API?

Reply:
This is intentional to be public and behind a debug_assertions gate. If you dig into the logic a bunch, global_matrix is actually not mutatable at all outside of the Transform system. It makes doing unit tests, tests cases and mocks a lot harder when you consider:
- All my rendering tests, I want to create a transform, tweak it a bit, then test it. 
- All rendering is supposed to work off global_matrix, which never get updated or modified except in TransformSystem.

This allows me to test things with Transform without using the entire amethyst test mockup framework to load transformsystem and tick it for me, when I am just doing these types of smaller tests.

I made it public because (as we discussed today) im using it in other external crates.